### DISCUSSION TITLE
Duplicate tracks

### DISCUSSION THREAD

**Discussion by lisadiez** [talmolab/sleap#1080] [Help!]
Hello,

We are trying to merge two slp projects together, but we're getting duplicate tracks:

<img width="1270" alt="duplicate_tracks" src="https://user-images.githubusercontent.com/116014547/207672972-b8500611-4865-4904-ac03-b1a505bfd934.png">

We first tried File --> Merge into Project... directly from the GUI and no conflicts emerged, but we still got the duplicate tracks.
We also tried with the following code (provided by Talmo a few months ago), but that didn't work either:

`import sleap
import sys

def merge(old_labels, new_labels):
    labels = sleap.load_file(old_labels)
    labels.remove_predictions()
    preds = sleap.load_file(new_labels, match_to=labels)

    # I don't know why doing this twice fixes it...
    merged, old_conflicts, new_conflicts = sleap.Labels.complex_merge_between(labels, preds, unify=False)
    merged, old_conflicts, new_conflicts = sleap.Labels.complex_merge_between(labels, preds, unify=False)
    print("Merged videos:", len(merged))
    print("Old labels conflicts:", len(old_conflicts))
    print("New labels conflicts:", len(new_conflicts))

    # Merge
    for video, video_preds in merged.items():
        for fidx, insts in video_preds.items():
            new_frame = False
            if (video, fidx) not in labels:
                lf = sleap.LabeledFrame(video=video, frame_idx=fidx)
                new_frame = True
            else:
                lf = labels[(video, fidx)]

            lf.instances.extend(insts)

            if new_frame:
                labels.append(lf)

    print("Tracks after merging:")
    print(labels.tracks)
    
    merged_filename = old_labels + ".merged.slp"
    labels.save(merged_filename)
    print("Saved:", merged_filename)
    
if __name__ == "__main__":
    if len(sys.argv) != 3:
        print("Call this script with syntax: 'python merge.py labels.slp predictions.slp'")
    merge(sys.argv[1], sys.argv[2])`

<img width="572" alt="terminal" src="https://user-images.githubusercontent.com/116014547/207673377-d1d82ce8-2dea-48a8-b45e-b1512441196d.png">

We're using v1.2.9.

Here's a folder with the videos and the two slp files: https://drive.google.com/drive/folders/1OPLs3iyld6Ib3iJYUKKxhj4a_h1mCgWM?usp=sharing

Do you have a workaround for this? 

Thank you!!
Lisa

**Comment by talmo** [comment]
FYI: we're tracking this as an issue in #1083. The temporary fix is to use something like [the merging notebook we made the other day](https://colab.research.google.com/drive/13DAiPiLq4_8suZOlPoD67ReIJzjN8oCW?usp=sharing).
